lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -173,7 +174,11 @@ def warn_about_negative_false_positive(expression)
         def expected_error
           case @expected_message
           when nil
-            description_of(@expected_error)
+            if RSpec::Support.is_a_matcher?(@expected_error)
+              "Exception with #{description_of(@expected_error)}"
```

Comment:
I guess we don't differentiate here between a matcher matching on the exception object vs message?  Seems like we'd want a different message for those cases.